lib/presentation/shared_widgets/phone_input.dart

```diff
@@ -0,0 +1,190 @@
+import 'package:collaction_app/presentation/authentication/utils/countries.dart';
+import 'package:collaction_app/presentation/shared_widgets/country_search_dialog.dart';
+import 'package:collaction_app/presentation/themes/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:phone_number/phone_number.dart';
+
+/// Phone input with country search and selection
+class PhoneInput extends StatefulWidget {
+  /// Callback for realtime input validation
+  final Function(bool)? isValid;
+
+  /// Callback with generated phone number
+  final Function(PhoneResponse)? onChange;
+
+  final PhoneResponse? phone;
+
+  const PhoneInput({Key? key, this.isValid, this.onChange, this.phone})
+      : super(key: key);
+
+  @override
+  _PhoneInputState createState() => _PhoneInputState();
+}
+
+class _PhoneInputState extends State<PhoneInput> {
+  Country? _selected;
+
+  final plugin = PhoneNumberUtil();
+  final _phoneNumberController = TextEditingController();
+  bool validatedNumber = false;
+
+  @override
+  void initState() {
+    if (widget.phone != null) {
+      final country =
+          countries.where((country) => country.code == widget.phone?.country);
+
+      /// Strip country
+      if (country.isNotEmpty) {
+        _selected = country.first;
+      } else {
+        _selected = countries.where((country) => country.code == "NL").first;
+      }
+
+      /// Strip phone
+      final contact = widget.phone?.contact.split(" ");
+
+      if (contact?.isNotEmpty == true && contact?.length == 2) {
+        _phoneNumberController.text = contact?.last ?? "";
+      }
+    } else {
+      _selected = countries.where((country) => country.code == "NL").first;
+    }
+
+    _phoneNumberController.addListener(() {
+      if (_phoneNumberController.value.text.isNotEmpty) {
+        setState(() {
+          _validatePhone(_selected!, _phoneNumberController.value.text);
+        });
+      }
+    });
+    super.initState();
```

Comment:
We should move super.initState(); to the first line of the initState method, it should always be called first.